canopy_driver_mod.F90

```diff
@@ -306,12 +313,37 @@ SUBROUTINE CANOPY_ZPD( ZHC, FCLAI, UBZREF, Z0GHC, &
 
         ! Calculate zero-plane displacement height, d/h (Eq. 15 in Massman et al. 2017):
         drag    = CDRAG*PAI
-        ustrmod = UBZREF*(0.38 - (0.38 + (vonk/log(Z0GHC)))*exp(-1.0*(15.0*drag)))
-        cstress = (2.0*(ustrmod**2.0))/(UBZREF**2.0)
+        ! Get first estimate of zpd and z0 to perform initial log wind profile and ustar calculation
+        ! Holmes JD. Wind Loading of Structures. 3rd ed. Boca Raton, Florida: CRC Press; 2015.
+        ! First assign general z0 dependent on vegetation type and then estimate zpd = 0.75*FCH from Holmes
+        if (LU_OPT .eq. 0) then !VIIRS LU types
+            !approx/average vegtype mapping to Massman et al. forest types
+            if (VTYPE .ge. 1 .and. VTYPE .le. 2) then !VIIRS Cat 1-2/Evergreen Needleleaf & Broadleaf
+                z0_set  = 1.0_rk
+            end if
+            if (VTYPE .ge. 3 .and. VTYPE .le. 5) then !VIIRS Cat 3-5/Deciduous Needleleaf, Broadleaf, Mixed Forests
+                z0_set = 1.0_rk
+            end if
+            if ((VTYPE .ge. 6 .and. VTYPE .le. 10) .or. VTYPE .eq. 12 ) then !VIIRS Cat 6-10 or 12/Shrubs, Croplands, and Grasses
+                z0_set = 0.1_rk
+            end if
```

Comment:
@zmoon Good point.  Right now we necessarily don't want to stop model if the VTYPE is not one of the supported mapping types as we will have numerous grid cells with other types. We  just do not want to calculate here or where the canopy shape parameters are set above in canopy_parm.  Do you have any recommendations on how to do this better?

Reply:
Ah, I also have the same LU_OPT and VTYPE checks before all this in the canopy_calcs, so this controls doing anything first based on LU and VTYPE before teh canopy driver routines:  https://github.com/noaa-oar-arl/canopy-app/blob/develop/canopy_calcs.F90#L56

I can do the if...else...if statemetns as you suggest here in canopy_driver as additional checks

Reply:
@zmoon If OK, I will be the if..then..else up in the canopy_calcs as it won't even reach these parts of the calculations in canopy_driver to do anything because a valid LU_OPT and VTYPE must be mapped, so they can stay the same here I think.  I could just put a warning statement higher up in canopy_calcs to alert grid cells without mapped VTYPES.  

Also, I think I should rename this module to canopy_profile or something else besides canopy_driver as its a bit misleading if you agree.

Reply:
> Also, I think I should rename this module to canopy_profile or something else besides canopy_driver as its a bit misleading if you agree.

Sure if you like. Maybe could move the dx calculation out in that case, to utils or something.

Reply:
@zmoon OK, great.  I have changed canopy_driver to  canopy_profile and moved dxcalc out to its own module [here](https://github.com/noaa-oar-arl/canopy-app/pull/34/commits/41e9f8039047860083734421bacf2062823ea9ca).  Also made a quick bug fix change to when calcdx occurs during both ifcanwind and ifcanwaf.